data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java

```diff
@@ -258,6 +258,10 @@ private void handleFailures(final AccumulatingBulkRequest<BulkOperationWrapper,
 
     private BulkOperationRequestResponse handleRetry(final AccumulatingBulkRequest request, final BulkResponse response, int retryCount) throws InterruptedException {
         final AccumulatingBulkRequest<BulkOperationWrapper, BulkRequest> bulkRequestForRetry = createBulkRequestForRetry(request, response);
+        if (bulkRequestForRetry.getOperationsCount() == 0) {
```

Comment:
Can you give some insight what is this getOperationsCount ?

Reply:
It is the number of operations in the bulk request. If this operation fails (for any reason), we will retry. But if the initial request has 0 operations, then there is no need to retry and no reason to write to DLQ when we fail to write.